### PR TITLE
docs(rpc): correct help for protx legacy versions

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented

Help text for protx legacy versions were adjusted.
## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge; seems fine

**Review by thephez — APPROVED** [review]
Concept ACK. Didn't test.

**Review by thephez — CHANGES_REQUESTED** [review]
Just kidding... :see_no_evil: Felt guilty about not testing.
The comparison is inverted in at least some cases.
image

Also, `protx register` / `protx register_legacy` don't seem to be covered. Should they be?

File: src/rpc/evo.cpp [hunk 1]
```diff
@@ -339,9 +339,13 @@ static std::string SignAndSendSpecialTx(const JSONRPCRequest& request, const CMu
     return sendrawtransaction(sendRequest).get_str();
 }
 
-static void protx_register_fund_help(const JSONRPCRequest& request)
+static void protx_register_fund_help(const JSONRPCRequest& request, bool legacy)
 {
-    RPCHelpMan{"protx register_fund",
+    std::string rpc_name = legacy ? "register_fund" : "register_fund_legacy";
```
Comment (thephez) [inline src/rpc/evo.cpp:344]:
```suggestion
    std::string rpc_name = legacy ? "register_fund_legacy" : "register_fund";
```

File: src/rpc/evo.cpp [hunk 2]
```diff
@@ -366,14 +370,18 @@ static void protx_register_fund_help(const JSONRPCRequest& request)
...
+            HelpExampleCli("protx",  rpc_example)
         },
     }.Check(request);
 }
 
-static void protx_register_help(const JSONRPCRequest& request)
+static void protx_register_help(const JSONRPCRequest& request, bool legacy)
 {
-    RPCHelpMan{"protx register",
+    std::string rpc_name = legacy ? "register" : "register_legacy";
```
Comment (thephez) [inline src/rpc/evo.cpp:380]:
```suggestion
    std::string rpc_name = legacy ? "register_legacy" : "register";
```

File: src/rpc/evo.cpp [hunk 3]
```diff
@@ -397,14 +405,18 @@ static void protx_register_help(const JSONRPCRequest& request)
...
+            HelpExampleCli("protx", rpc_example)
         },
     }.Check(request);
 }
 
-static void protx_register_prepare_help(const JSONRPCRequest& request)
+static void protx_register_prepare_help(const JSONRPCRequest& request, bool legacy)
 {
-    RPCHelpMan{"protx register_prepare",
+    std::string rpc_name = legacy ? "register_prepare" : "register_prepare_legacy";
```
Comment (thephez) [inline src/rpc/evo.cpp:415]:
```suggestion
    std::string rpc_name = legacy ? "register_prepare_legacy" : "register_prepare";
```

**Review by thephez — COMMENTED** [review]
I think this is the correct order for these...

File: src/rpc/evo.cpp [hunk 4]
```diff
@@ -87,14 +87,25 @@ static RPCArg GetRpcArg(const std::string& strParamName)
         {"operatorPubKey_register",
             {"operatorPubKey_register", RPCArg::Type::STR, RPCArg::Optional::NO,
                 "The operator BLS public key. The BLS private key does not have to be known.\n"
-                "It has to match the BLS private key which is later used when operating the masternode."}
+                "It has to match the BLS private key which is later used when operating the masternode.\n"}
```
Comment (thephez) [inline src/rpc/evo.cpp:90]: Looks like this adds an extra blank line.
```suggestion
                "It has to match the BLS private key which is later used when operating the masternode."}
```

**Review by thephez — DISMISSED** [review]
LGTM :+1:

**Review by thephez — APPROVED** [review]
:+1: Good catch

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**